plugins/tiledb/io/TileDBWriter.cpp

```diff
@@ -137,75 +156,133 @@ tiledb::Attribute createAttribute(const tiledb::Context& ctx,
 }
 
 
-tiledb::Filter createFilter(const tiledb::Context& ctx, const std::string& name)
+std::unique_ptr<tiledb::Filter> createFilter(const tiledb::Context& ctx, const NL::json& opts)
 {
+    std::unique_ptr<tiledb::Filter> filter;
+    std::string name = opts["compression"];
+
     if (name.empty())
-        return tiledb::Filter(ctx, TILEDB_FILTER_NONE);
+        filter.reset(new tiledb::Filter(ctx, TILEDB_FILTER_NONE));
     else if (name == "gzip")
-        return tiledb::Filter(ctx, TILEDB_FILTER_GZIP);
+        filter.reset(new tiledb::Filter(ctx, TILEDB_FILTER_GZIP));
     else if (name == "zstd")
-        return tiledb::Filter(ctx, TILEDB_FILTER_ZSTD);
+        filter.reset(new tiledb::Filter(ctx, TILEDB_FILTER_ZSTD));
     else if (name == "lz4")
-        return tiledb::Filter(ctx, TILEDB_FILTER_LZ4);
+        filter.reset(new tiledb::Filter(ctx, TILEDB_FILTER_LZ4));
     else if (name == "rle")
-        return tiledb::Filter(ctx, TILEDB_FILTER_RLE);
+        filter.reset(new tiledb::Filter(ctx, TILEDB_FILTER_RLE));
     else if (name == "bzip2")
-        return tiledb::Filter(ctx, TILEDB_FILTER_BZIP2);
+        filter.reset(new tiledb::Filter(ctx, TILEDB_FILTER_BZIP2));
     else if (name == "double-delta")
-        return tiledb::Filter(ctx, TILEDB_FILTER_DOUBLE_DELTA);
+        filter.reset(new tiledb::Filter(ctx, TILEDB_FILTER_DOUBLE_DELTA));
+    else if (name == "bit-width-reduction")
+        filter.reset(new tiledb::Filter(ctx, TILEDB_FILTER_BIT_WIDTH_REDUCTION));
+    else if (name == "bit-shuffle")
+        filter.reset(new tiledb::Filter(ctx, TILEDB_FILTER_BITSHUFFLE));
+    else if (name == "byte-shuffle")
+        filter.reset(new tiledb::Filter(ctx, TILEDB_FILTER_BYTESHUFFLE));
+    else if (name == "positive-delta")
+        filter.reset(new tiledb::Filter(ctx, TILEDB_FILTER_POSITIVE_DELTA));
     else
         throw tiledb::TileDBError("Unable to parse compression type: " + name);
+
+    if (opts.count("compression_level") > 0)
+        filter->set_option(TILEDB_COMPRESSION_LEVEL, opts["compression_level"].get<int>());
```

Comment:
If the actual JSON type isn't an `int`, these will throw an exception IIRC.

Reply:
is it ok for this to throw an exception? It is expecting an `int` or would you rather the code checks and throws its own exception?